### PR TITLE
KAFKA-5746; Fix conversion count computed in `downConvert`

### PR DESCRIPTION
**Pull request by ijuma**
It should be the number of records instead of the
number of batches.

A few additional clean-ups:
- Minor renames
- Removed unused variable
- Some test fixes
- Ignore a flaky test

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @rajinisivaram.

File: clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java [hunk 1]
```diff
@@ -18,15 +18,15 @@
 
 public class RecordsProcessingStats {
 
-    public static final RecordsProcessingStats EMPTY = new RecordsProcessingStats(0L, 0L, -1);
+    public static final RecordsProcessingStats EMPTY = new RecordsProcessingStats(0L, 0, -1);
 
     private final long temporaryMemoryBytes;
-    private final long conversionCount;
+    private final int numRecordsConverted;
```
Comment (tedyu) [inline clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java:24]: Shouldn't this be of long type ?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java:24]: I don't think so. What would be the reason?

Comment (tedyu) [reply clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java:24]: The for loop structure involving line 112 of AbstractRecords.java doesn't change.
Increment by 1 is changed to increment by builder.numRecords().
Meaning the final outcome would be bigger than prior to change.

Hence I think this should keep the long type.

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java:24]: The point was that `long` was unnecessary in the first place. We can never read more than `2 GB` from the log at a time:

```scala
def read(startOffset: Long, maxLength: Int, maxOffset: Option[Long] = None, minOneMessage: Boolean = false,
           isolationLevel: IsolationLevel): FetchDataInfo = {
```

Comment (tedyu) [reply clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java:24]: Good to know.

**Review by rajinisivaram — APPROVED** [review]
LGTM

**Pull request closed without merging**